cmds/boot/uinit/main.go

```diff
@@ -28,6 +29,47 @@ var defaultBootsequence = [][]string{
 	{"localboot", "-grub"},
 }
 
+func getSystemFWVersion() (string, error) {
+	entryData, tableData, err := getSmbiosData()
+	if err != nil {
+		return "", err
+	}
+	si, err := smbios.ParseInfo(entryData, tableData)
+	if err != nil {
+		log.Printf("error parsing data: %v", err)
+		return "", err
+	}
+	t0, err := si.GetBIOSInformation()
+	if err != nil {
+		log.Printf("error Geting BIOS Information: %v", err)
+		return "", err
+	}
+	return t0.Version, nil
+}
+
+func runIpmiCommands() {
```

Comment:
I'm sorry to add one more request to this, but let's get it right from the start. Go convention is to upcase all letters in an acronym used in a function name, unlike UEFI: So:
TCP not Tcp (I know UEFI does Tcp but this is Go)
IPMI not Ipmi
so this function would be better if named
runIPMICommands

The *package* name ipmi is fine. but functions and variables should be IPMI, not Ipmi.

Reply:
No problem, updated.